### ISSUE TITLE
the accuracy of implement

### ISSUE THREAD

**Issue by zqy-happy** [huggingface/peft#589]
I ran this code using the entire Food101 dataset and found that the model trained with the original code achieved only around 0.06 accuracy on the total validation set. However, when I used the entire training set of Food101 with learning rates of 0.005, 0.001, and 0.0005, the best accuracy I obtained was approximately 0.906. Has anyone else encountered a similar issue?

**Comment by pacman100** [comment]
Hello @zqy-happy, I'm unable to understand the issue here. Could you please give a lot more context following the issue template for us to help you?

**Comment by zqy-happy** [comment]
> Hello @zqy-happy, I'm unable to understand the issue here. Could you please give a lot more context following the issue template for us to help you?

Thank you! I'm glad to receive your reply. So, first of all, I ran the entire code from the GitHub repository "https://github.com/huggingface/peft/tree/main/examples/image_classification/image_classification_peft_lora.ipynb". The dataset part was loaded as shown in the project using the code "dataset = load_dataset("food101", split="train[:5000]")". I obtained results that were essentially the same as those in the repository.

However,I decided to reload the entire Food101 dataset using the following code: "dataset = load_dataset("food101", cache_dir="./datasets")" and "train_ds = dataset['train'] , 
val_ds = dataset['validation']". After reloading the dataset, I performed testing again on the fine-tuned model using the code "trainer.evaluate(val_ds)". The results I obtained were around 0.06.

Furthermore, I used similar code but with the complete Food101 dataset and experimented with different learning rates such as 0.005, 0.001, and 0.0005. The following are the results I obtained:
Learning Rate: 0.005

Evaluation Loss: 0.4101414680480957
Evaluation Accuracy: 0.8912079207920792
Evaluation Runtime: 332.0338 seconds
Evaluation Samples per Second: 76.046
Evaluation Steps per Second: 0.596
Epoch: 29.0

Learning Rate: 0.001

Evaluation Loss: 0.3431929647922516
Evaluation Accuracy: 0.9047524752475248
Evaluation Runtime: 209.8496 seconds
Evaluation Samples per Second: 120.324
Evaluation Steps per Second: 0.944
Epoch: 34.0


Learning Rate: 0.0005

Evaluation Loss: 0.3281923234462738
Evaluation Accuracy: 0.9061782178217822
Evaluation Runtime: 224.6711 seconds
Evaluation Samples per Second: 112.387
Evaluation Steps per Second: 0.881
Epoch: 36.0

I would like to know if there were any issues during this process.

**Comment by pacman100** [comment]
cc @sayakpaul who has a lot more context on this

**Comment by sayakpaul** [comment]
I am unable to reproduce this issue. 

Could you maybe provide a Colab Notebook for this? You can use this one: https://github.com/huggingface/peft/blob/main/examples/image_classification/image_classification_peft_lora.ipynb

**Comment by zqy-happy** [comment]
> I am unable to reproduce this issue.
> 
> Could you maybe provide a Colab Notebook for this? You can use this one: https://github.com/huggingface/peft/blob/main/examples/image_classification/image_classification_peft_lora.ipynb

I apologize for the delay in my response. I am not familiar with Colab, but I have a repository where I execute code on a Linux platform with GPU, which can provide more detailed information. 

Moreover, I realize that I may not have articulated my intention clearly！ Simply put, I would like to inquire about the baseline results achieved by others, including yourself, when fine-tuning a Vision Transformer (ViT) using Lora on the entire 'Food101' dataset or any other datasets.

Thank you.

**Comment by sayakpaul** [comment]
If you run the Colab notebook, it should give at least >80% accuracy (top-1). So, for now, I am going to close the issue.
